deploy/zookeeper/quick-start-volume-emptyDir/zookeeper-3-nodes.yaml

```diff
@@ -96,41 +96,77 @@ spec:
-          --snap_retain_count=3 \
-          --purge_interval=12 \
-          --max_session_timeout=40000 \
-          --min_session_timeout=4000 \
-          --log_level=INFO"
+            - |
+              SERVERS=3 &&
+              HOST=`hostname -s` &&
+              DOMAIN=`hostname -d` &&
+              CLIENT_PORT=2181 &&
+              SERVER_PORT=2888 &&
+              ELECTION_PORT=3888 &&
+              {
+                echo "clientPort=${CLIENT_PORT}"
+                echo 'tickTime=2000'
+                echo 'initLimit=30000'
+                echo 'syncLimit=10'
+                echo 'maxClientCnxns=2000'
+                echo 'maxSessionTimeout=60000000'
+                echo 'dataDir=/data'
+                echo 'dataLogDir=/datalog'
+                echo 'autopurge.snapRetainCount=10'
+                echo 'autopurge.purgeInterval=1'
+                echo 'preAllocSize=131072'
+                echo 'snapCount=3000000'
+                echo 'leaderServes=yes'
+                echo 'standaloneEnabled=true'
+                echo '4lw.commands.whitelist=stat, ruok, conf, isro'
+              } > /conf/zoo.cfg &&
+              {
+                echo "zookeeper.root.logger=CONSOLE"
+                echo "zookeeper.console.threshold=INFO"
+                echo "log4j.rootLogger=\${zookeeper.root.logger}"
+                echo "log4j.appender.CONSOLE=org.apache.log4j.ConsoleAppender"
+                echo "log4j.appender.CONSOLE.Threshold=\${zookeeper.console.threshold}"
+                echo "log4j.appender.CONSOLE.layout=org.apache.log4j.PatternLayout"
+                echo "log4j.appender.CONSOLE.layout.ConversionPattern=%d{ISO8601} [myid:%X{myid}] - %-5p [%t:%C{1}@%L] - %m%n"
+              } > /conf/log4j.properties &&
+              echo 'JVMFLAGS="-Xms128M -Xmx1G -XX:+UseParNewGC -XX:+UseConcMarkSweepGC -XX:+CMSParallelRemarkEnabled"' > /conf/java.env &&
+              if [[ $HOST =~ (.*)-([0-9]+)$ ]]; then
+                  NAME=${BASH_REMATCH[1]}
+                  ORD=${BASH_REMATCH[2]}
+              else
+                  echo "Fialed to parse name and ordinal of Pod"
+                  exit 1
+              fi &&
+              export MY_ID=$((ORD+1)) &&
+              echo $MY_ID > $ZOO_DATA_DIR/myid &&
+              if [[ $SERVERS -gt 1 ]]; then
+                for (( i=1; i<=$SERVERS; i++ )); do
+                    echo "server.$i=$NAME-$((i-1)).$DOMAIN:$SERVER_PORT:$ELECTION_PORT" >> /conf/zoo.cfg;
+                done
+              fi &&
+              chown -Rv zookeeper "$ZOO_DATA_DIR" "$ZOO_DATA_LOG_DIR" "$ZOO_LOG_DIR" "$ZOO_CONF_DIR" &&
+              zkServer.sh start-foreground
           readinessProbe:
             exec:
               command:
-                - sh
+                - bash
                 - -c
-                - "zookeeper-ready 2181"
+                - "OK=$(echo ruok | nc 127.0.0.1 2181); if [[ \"$OK\" == \"imok\" ]]; then exit 0; else exit 1; fi"
```

Comment:
May be this may/shoult be used in combination with / replaced by ```./zkServer.sh status``` ?

Reply:
please look at https://github.com/31z4/zookeeper-docker/blob/master/3.5.6/Dockerfile#L3 this environment variables defined as ENV section

Reply:
`zkServer.sh status` use `srvr` command which run complex action and show "details" statistics
but `ruok` check only is ZK running and accept connections and not in error state 

i compare sources for srvr command
https://github.com/apache/zookeeper/blob/master/zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/Commands.java#L551

and ruok command
https://github.com/apache/zookeeper/blob/master/zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/Commands.java#L490

so, I think for kubernetes liveness probe it's ok to use `ruok`, but for readiness probe better solution `zkServer.sh status`

I will apply these changes shortly

Reply:
cause kubernetes run readiness and liveness probe every 10 seconds


```
2020-01-15 13:46:48,299 [myid:1] - INFO  [NIOWorkerThread-1:NIOServerCnxn@518] - Processing srvr command from /127.0.0.1:57864
2020-01-15 13:46:49,498 [myid:1] - INFO  [NIOWorkerThread-2:NIOServerCnxn@518] - Processing ruok command from /127.0.0.1:57872
2020-01-15 13:46:57,788 [myid:1] - INFO  [NIOWorkerThread-1:NIOServerCnxn@518] - Processing srvr command from /127.0.0.1:57930
2020-01-15 13:46:59,235 [myid:1] - INFO  [NIOWorkerThread-2:NIOServerCnxn@518] - Processing ruok command from /127.0.0.1:57940
2020-01-15 13:47:08,262 [myid:1] - INFO  [NIOWorkerThread-1:NIOServerCnxn@518] - Processing srvr command from /127.0.0.1:58006
2020-01-15 13:47:09,293 [myid:1] - INFO  [NIOWorkerThread-2:NIOServerCnxn@518] - Processing ruok command from /127.0.0.1:58012
```

I think better still use 'ruok' both for liveness and readiness 
cause `srvr` is much more complex command